src/connectionbenchmark/benchmarktaskdownload.cpp

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "benchmarktaskdownload.h"
+#include "constants.h"
+#include "leakdetector.h"
+#include "logger.h"
+#include "networkrequest.h"
+
+#include <QByteArray>
+
+namespace {
+Logger logger(LOG_MAIN, "BenchmarkTaskDownload");
+}
+
+BenchmarkTaskDownload::BenchmarkTaskDownload(const QString& fileUrl)
+    : BenchmarkTask(Constants::BENCHMARK_MAX_DURATION_DOWNLOAD),
+      m_fileUrl(fileUrl) {
+  MVPN_COUNT_CTOR(BenchmarkTaskDownload);
+}
+
+BenchmarkTaskDownload::~BenchmarkTaskDownload() {
+  MVPN_COUNT_DTOR(BenchmarkTaskDownload);
+}
+
+void BenchmarkTaskDownload::runInternal() {
+  logger.debug() << "Run download benchmark";
+
+  connect(this, &BenchmarkTask::stateChanged, this,
```

Comment:
I like this approach! Let's move it to the "extreme"! I would do this:

1. remove `runInternal` entirelly
2. move this `connect` in the CTOR of the `BenchmarkTaskDownload`